libs-scala/resources/src/main/scala/com/digitalasset/resources/TimerResourceOwner.scala

```diff
@@ -3,12 +3,27 @@
 
 package com.daml.resources
 
-import java.util.Timer
-
-import scala.concurrent.Future
+import java.util.{Timer, TimerTask}
+import scala.concurrent.{Future, Promise}
+import scala.util.Try
 
 class TimerResourceOwner[Context: HasExecutionContext](acquireTimer: () => Timer)
     extends AbstractResourceOwner[Context, Timer] {
   override def acquire()(implicit context: Context): Resource[Context, Timer] =
-    ReleasableResource(Future(acquireTimer()))(timer => Future(timer.cancel()))
+    ReleasableResource(Future(acquireTimer())) { timer =>
+      val timerCancelledPromise = Promise[Unit]()
+      Future(
+        timer.schedule(
```

Comment:
I would prefer a comment here which would copy-paste the contract of `timer.cancel` method which we are relying on (are we?), i.e.:

```
Note that calling this method from within the run method of a timer task that was invoked by this timer absolutely guarantees that the ongoing task execution is the last task execution that will ever be performed by this timer.
```